numpy/core/src/multiarray/ctors.c

```diff
@@ -422,6 +422,10 @@ copy_and_swap(void *dst, void *src, int itemsize, npy_intp numitems,
     }
 }
 
+static PyObject *
+_array_from_array_like(PyObject *op, PyArray_Descr *requested_dtype,
+                       npy_bool writeable, PyObject *context);
+
```

Comment:
`static` is fine, but then the implementation should also have the `static` attribute, and might need to move before any use of the function, depending on compiler. We usually use `static` for functions in the same file, and `NPY_NO_EXPORT` for functions that will be used across different files but not exported from the `*.so`

Reply:
I've changed `static` to `NPY_NO_EXPORT` to match `_array_from_buffer_3118`. I understand that this does not match the "used across different files" convention but at least it's consistent between similar function.